src/main/java/com/salesforce/dynamodbv2/mt/util/CachingAmazonDynamoDbStreams.java

```diff
@@ -592,36 +414,30 @@ public int hashCode() {
         }
     }
 
-    // logger instance
-    private static final Logger LOG = LoggerFactory.getLogger(CachingAmazonDynamoDbStreams.class);
+    private static class CachingGetRecordsResult {
+        private final List<Record> records;
+        private final CachingShardIterator nextShardIterator;
```

Comment:
doesn't need to be in this PR, but I'd also like to see stats of cache hits vs misses (how many stream callouts were made vs how much came off the cache), and given we have potential lock wait states, timing information for those wait states

Reply:
One of the things we were discussing before I left was defining some kind of metrics interface for mt-dynamo that users of the library could supply. Is something like that in progress as part of the metrics work you guys have been doing?

Reply:
I have exposed the 2 caches so can be provided by the client that is created the MT AmazonDynamoDB via the SharedTableBuilder.  I haven't defined an interface that calls back with metrics.  But I have on my personal list a desire to record consumed capacities and that will necessitate such an interface so I will likely get to it soon.